src/jdk.management/share/classes/com/sun/management/HotSpotDiagnosticMXBean.java

```diff
@@ -117,4 +118,54 @@ public interface HotSpotDiagnosticMXBean extends PlatformManagedObject {
      *     ManagementPermission("control").
      */
     public void setVMOption(String name, String value);
+
+    /**
+     * Generate a thread dump to the given file in the given format. The
+     * {@code outputFile} parameter must be an absolute path to a file that
+     * does not exist.
+     *
+     * <p> The thread dump will include output for all platform threads. It may
+     * include output for some or all virtual threads.
+     *
+     * @implSpec
+     * The default implementation throws {@code UnsupportedOperationException}.
+     *
+     * @apiNote
+     * The output file is required to be an absolute path as the MXBean may be
+     * accessed remotely from a tool or program with a different current working
+     * directory.
+     *
+     * @param  outputFile the path to the file to create
+     * @param  format the format to use
+     * @throws IllegalArgumentException if the file path is not absolute
+     * @throws IOException if the file already exists or an I/O exception is
+     *         thrown writing to the file
+     * @throws NullPointerException if either parameter is {@code null}
+     * @throws SecurityException
+     *         if a security manager is set and its {@link
+     *         SecurityManager#checkWrite(java.lang.String)} method denies write
+     *         access to the file or {@link java.lang.management.ManagementPermission
+     *         ManagementPermission("control")} is denied
+     * @since 19
```

Comment:
Maybe there ought to be an `@throws UnsupportedOperationException` here since that is what the default implementation of the method is supposed to do.

Reply:
Well spotted. The implSpec further up documents it but it should be in the throws list too.